VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -148,36 +184,49 @@ IOReturn VoodooI2CHIDDevice::getHIDDescriptorAddress() {
 void VoodooI2CHIDDevice::getInputReport() {
     IOBufferMemoryDescriptor* buffer;
     IOReturn ret;
-    unsigned char* report = (unsigned char *)IOMalloc(hid_descriptor.wMaxInputLength);
-
-    api->readI2C(report, hid_descriptor.wMaxInputLength);
+    int return_size;
+    unsigned char* report;
     
-    int return_size = report[0] | report[1] << 8;
+    if (I2C_TRYLOCK() == false) {
+        IOLog("%s::%s Skipping a HID read interrupt while other thread read/write I2C\n", getName(), name);
+        goto exit;
```

Comment:
`goto exit;` essentially means assigning `read_in_progress = false;`, which is wrong since another thread is reading/writing I2C and mutex is locked by that thread, not current thread. This could lead to undefined behavior.

I think we should `return;` here.